service/storage_service.cc

```diff
@@ -2530,6 +2539,9 @@ future<> topology_coordinator::run() noexcept {
 
     co_await _async_gate.close();
     co_await std::move(cdc_generation_publisher);
+  } catch (...) {
+    on_fatal_internal_error(slogger, format("raft topology: unhandled exception in topology_coordinator::run: {}", std::current_exception()));
+  }
```

Comment:
I do not like this approach. The code above already handled all exceptions. I is fine to use simple `noexcep` C++ keyword to mark a function as non throwing more for the documentation purposes then to actually terminate the program. Re-implementing `noexcep`  semantics manually just add noise the the code. If you want to asser you can do it in `raft_state_monitor_fiber` when waiting for `_topology_change_coordinator` feature which suppose to be non exceptional.